### PR TITLE
Add license endpoints and type

### PR DESCRIPTION
**Pull request by SaschaMann**
~WIP/Draft to avoid duplicate work, it's not yet finished~

### CODE REVIEW

File: src/GitHub.jl [hunk 1]
```diff
@@ -325,4 +325,16 @@ export
     create_release,
     releases
 
+############
+# Licenses #
+############
+
+include("licences/license.jl")
```
Comment (christopher-dG) [inline src/GitHub.jl:332]: You gotta choose one spelling and stick to it 😅 Looks like `license` is what GitHub uses.

Comment (SaschaMann) [reply src/GitHub.jl:332]: I tried so hard to stick to license :D Good catch :)

**Comment by SaschaMann** [comment]
I'm not familiar with the internals of this package, so I tried to copy existing files/tests. Please let me know if anything is wrong or missing.

File: src/licenses/license.jl [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    permissions::Union{Vector{String}, Nothing}
+    conditions::Union{Vector{String}, Nothing}
+    limitations::Union{Vector{String}, Nothing}
+    body::Union{String, Nothing}
+    featured::Union{Bool, Nothing}
+end
+
+License(spdx_id::AbstractString) = License(Dict("spdx_id" => spdx_id))
+
+namefield(license::License) = license.spdx_id
```
Comment (SaschaMann) [inline src/licenses/license.jl:21]: I'm not sure what the best choice is here, the obvious options would be `key`, `name`, or `spdx_id`. Some examples:

```json
[
  {
    "key": "mit",
    "name": "MIT License",
    "spdx_id": "MIT"
  },
  {
    "key": "lgpl-3.0",
    "name": "GNU Lesser General Public License v3.0",
    "spdx_id": "LGPL-3.0"
  },
  {
    "key": "mpl-2.0",
    "name": "Mozilla Public License 2.0",
    "spdx_id": "MPL-2.0"
  },
  {
    "key": "agpl-3.0",
    "name": "GNU Affero General Public License v3.0",
    "spdx_id": "AGPL-3.0"
  },
  {
    "key": "unlicense",
    "name": "The Unlicense",
    "spdx_id": "Unlicense"
  },
  {
    "key": "apache-2.0",
    "name": "Apache License 2.0",
    "spdx_id": "Apache-2.0"
  },
  {
    "key": "gpl-3.0",
    "name": "GNU General Public License v3.0",
    "spdx_id": "GPL-3.0"
  }
]
```

Comment (christopher-dG) [reply src/licenses/license.jl:21]: I think it's only super important when the field is used as an identifier in other API calls. Seems like there aren't any here.

Comment (SaschaMann) [reply src/licenses/license.jl:21]: A call to `licence` will take it as identifier. Afaict the endpoint takes both `key` and `spdx_id` but not `name`.

Comment (christopher-dG) [reply src/licenses/license.jl:21]: Ah okay. I think the SPDX ID is fine then!

**Comment by SaschaMann** [comment]
Is there anything else I need to do for this to be mergeable?

**Review by christopher-dG — APPROVED** [review]
LGTM, I'm gonna let someone else a bit more familiar with this package merge though.

**Comment by KristofferC** [comment]
LGTM as well

**Pull request merged**